File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

func ToRNA(dna string) string {
	var rna []rune
	for _, c := range dna {
		if c == 'G' {
			rna = append(rna, 'C')
		} else if c == 'C' {
			rna = append(rna, 'G')
		} else if c == 'T' {
			rna = append(rna, 'A')
		} else if c == 'A' {
			rna = append(rna, 'U')
		}
	}
	return string(rna)
}

File: solutions/c/rna-transcription/3/rna_transcription.c
#include "rna_transcription.h"
#include <stdlib.h>
#include <string.h>

char *to_rna(const char *dna) {

    char *rna = malloc(strlen(dna) + 1);
    unsigned long i = 0;
    for (i = 0; i < strlen(dna); i++) {
        switch (dna[i]) {
            case 'G':
                rna[i] = 'C';
                break;
            case 'C':
                rna[i] = 'G';
                break;
            case 'T':
                rna[i] = 'A';
                break;
            case 'A':
                rna[i] = 'U';
                break;
        }
    }
    rna[i] = '\0';
    return rna;
}
